opm/grid/cpgrid/CpGrid.cpp

```diff
@@ -238,20 +240,40 @@ CpGrid::scatterGrid(EdgeWeightMethod method,
         return std::make_pair(false, std::vector<std::pair<std::string,bool> >());
     }
 
-    if (data_.size() > 1)
-    {
-        if (comm().rank() == 0)
-        {
-            OPM_THROW(std::logic_error, "Loadbalancing a grid with local grid refinement is not supported, yet.");
+#if HAVE_MPI
+    bool validLevel = (level>-1) && (level <= maxLevel());
+    // If level == -1, leaf grid view should be distributed (with/without LGRs).
+    // - without LGRs: leaf grid view coincides with level zero grid. Supported.
+    // - with LGRs: not supported yet. Throw in that case.
+    int selectedLevel = validLevel? level : 0;
```

Comment:
Is this correct? If I pass -1 for the level then validLevel is false and selectedLevel seems to become 0.
It might also be that it is correct but the variable name does not reflect what the variable is used for. Maybe we should  decide whether to throw based on the variable level to make the code more explicit?

Reply:
> If I pass -1 for the level then validLevel is false and selectedLevel seems to become 0.

Indeed, that's correct. I agree that the 'selectedLevel' variable could create some confusion. Let's decide the throw depending on level. 

The idea behind "selectedLevel" is that, for CpGrid with/without LGRs you can always - now - distribute level zero grid. For CpGrid with LGRs, this PR paves the way to allow distribution of refined grids (level>0) but it is not fully supported yet. Communication needs to be fixed/refactored. 
In conclusion, in this PR the distributed level grid is always zero.